guides/aws/mte-relay-server-container-usage-guide.md

```diff
@@ -136,24 +138,110 @@ _The customer is required to create the following keys. It is recommended that t
+  - example: "/some_route_that_is_not_secret"
+- `MTE_ROUTES`
+  - A list of routes that will be MTE encoded/decoded. If this optional property is included, only the routes listed will be MTE encoded/decoded, and any routes not listed here or in `PASS_THROUGH_ROUTES` will 404. If this optional property is not included, all routes not listed in `PASSTHROUGH_ROUTES` will be MTE encoded/decoded.
+- `CORS_METHODS`
+  - A list of HTTP methods that will be allowed to make cross-origin requests to the server.
+  - Default: `GET, POST, PUT, DELETE`.
+  - Note: `OPTIONS` and `HEAD` are always allowed.
+- `HEADERS`
+  - An object of headers that will be added to all request/responses.
+- `MAX_POOL_SIZE`
+  - The number of encoder objects and decoder objects held in a pool. A larger pool will consume more memory, but it will also handle more traffic more quickly. This number is applied to all four pools; the MTE Encoder, MTE Decoder, MKE Encoder, and MKE Decoder pools.
+  - Default: `25`
+### YAML Configuration Examples
+[AWS Task Definition Parameters](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/taskdef-envfiles.html)
+#### Minimal Configuration Example
+```yaml
+upstream: https://api.my-company.com
+clientIdSecret: 2DkV4DDabehO8cifDktdF9elKJL0CKrk
+corsOrigins:
+  - https://www.my-company.com
+  - https://dashboard.my-company.com
+redisURL: redis://10.0.1.230:6379
+```
 
-Please see [MTE Relay Server Docs](https://github.com/Eclypses/mte-relay-server/blob/master/README.md#configuration-options) for more info.
+#### Full Configuration Example
+
+```yaml
+upstream: https://api.my-company.com
+clientIdSecret: 2DkV4DDabehO8cifDktdF9elKJL0CKrk
+corsOrigins:
+  - https://www.my-company.com
+  - https://dashboard.my-company.com
+redisURL: redis://10.0.1.230:6379
+port: 3000
+debug: true
+passThroughRoutes:
+  - /health
+  - /version
+mteRoutes:
+  - /api/v1/*
+  - /api/v2/*
+corsMethods:
+  - GET
+  - POST
+  - DELETE
+headers:
+  x-service-name: mte-relay
+maxPoolSize: 10
+```
+
+### ENV Configuration Examples
+- [AWS Task Definition Environment File](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/use-environment-file.html)
+- [AWS Retrieve Secrets Manager Secrets](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/secrets-envvar-secrets-manager.html)
+#### Minimal Configuration Example
+```env
+UPSTREAM=https://api.my-company.com
+CLIENT_ID_SECRET=2DkV4DDabehO8cifDktdF9elKJL0CKrk
+CORS_ORIGINS=https://www.my-company.com,https://dashboard.my-company.com
+REDIS_URL=redis://10.0.1.230:6379
+```
+
+#### Full Configuration Example
+
+```env
+UPSTREAM=https://api.my-company.com
+CLIENT_ID_SECRET=2DkV4DDabehO8cifDktdF9elKJL0CKrk
+CORS_ORIGINS=https://www.my-company.com,https://dashboard.my-company.com
+REDIS_URL=redis://10.0.1.230:6379
+PORT: 3000
+DEBUG: true
+PASSTHROUGH_ROUTES=/health,/version
+MTE_ROUTES=/api/v1/*,/api/v2/*
+CORS_METHODS=GET,POST,DELETE
+HEADERS=x-service-name:mte-relay
```

Comment:
This needs to be a JSON string that can be parsed into an object. Example: 
`HEADERS='{"x-service-name":"mte-relay"}'`

Reply:
Resolved